nixos/modules/services/networking/wasabibackend.nix

```diff
@@ -0,0 +1,145 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.wasabibackend;
+
+  inherit (lib) mkEnableOption mkIf mkOption optionalAttrs types;
+
+	confOptions = {
+			BitcoinRpcConnectionString = "${cfg.rpc.user}:${cfg.rpc.password}";
+	} // optionalAttrs (cfg.network == "mainnet") {
+			Network = "Main";
+			MainNetBitcoinP2pEndPoint = "${cfg.endpoint.ip}:${toString cfg.endpoint.port}";
+			MainNetBitcoinCoreRpcEndPoint = "${cfg.rpc.ip}:${toString cfg.rpc.port}";
+	} // optionalAttrs (cfg.network == "testnet") {
+			Network = "TestNet";
+			TestNetBitcoinP2pEndPoint = "${cfg.endpoint.ip}:${toString cfg.endpoint.port}";
+			TestNetBitcoinCoreRpcEndPoint = "${cfg.rpc.ip}:${toString cfg.rpc.port}";
+	} // optionalAttrs (cfg.network == "regtest") {
+			Network = "RegTest";
+			RegTestBitcoinP2pEndPoint = "${cfg.endpoint.ip}:${toString cfg.endpoint.port}";
+			RegTestBitcoinCoreRpcEndPoint = "${cfg.rpc.ip}:${toString cfg.rpc.port}";
+	};
+
+	configFile = pkgs.writeText "wasabibackend.conf" (builtins.toJSON confOptions);
+
+in {
+
+  options = {
+
+    services.wasabibackend = {
+      enable = mkEnableOption "Wasabi backend service";
+
+      dataDir = mkOption {
+        type = types.path;
+        default = "/var/lib/wasabibackend";
+        description = "The data directory for the Wasabi backend node.";
+      };
+
+			customConfigFile = mkOption {
+				type = types.nullOr types.path;
+				default = null;
+				description = "Defines the path to a custom configuration file that is copied to the user's directory. Overrites any config options.";
+			};
+
+      network = mkOption {
+        type = types.enum [ "mainnet" "testnet" "regtest" ];
+        default = "mainnet";
+        description = "The network to use for the Wasabi backend service.";
+      };
+
+			endpoint = {
+				ip = mkOption {
+					type = types.str;
+					default = "127.0.0.1";
+					description = "IP address for P2P connection to bitcoind.";
+				};				
+				port = mkOption {
+					type = types.port;
+					default = 18333;
```

Comment:
The port numbers are for testnet (afaik) but the default network is mainnet. Would it make more sense to use 8332 & 8333?

Reply:
Good idea!